src/lib.rs

```diff
@@ -10,15 +10,59 @@ use wasm_bindgen::JsCast;
 #[global_allocator]
 static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
 
-#[wasm_bindgen]
+#[derive(Clone, Copy, PartialEq, Eq)]
```

Comment:
`derive` 😲

Reply:
Derive is _great_, it implements what I'd call "the most obvious implementation" of a trait for your type. So unlike Java where (unless you use compiler plugins I think) you have to describe how `==` should work by hand, or JavaScript where you can have any kind of equality you want as long as it's referential equality, in Rust you get the best of both worlds. You get to define basically everything about how your type should work, but if you just want the obvious implementation, you can usually just `derive` it.